pkg/render/node.go

```diff
@@ -1220,6 +1245,27 @@ func (c *nodeComponent) nodeEnvVars() []corev1.EnvVar {
 	if c.bpfDataplaneEnabled() {
 		nodeEnv = append(nodeEnv, corev1.EnvVar{Name: "FELIX_BPFENABLED", Value: "true"})
 	}
+	if c.vppDataplaneEnabled() {
+		nodeEnv = append(nodeEnv, corev1.EnvVar{
+			Name:  "FELIX_USEINTERNALDATAPLANEDRIVER",
+			Value: "false",
+		}, corev1.EnvVar{
+			Name:  "FELIX_DATAPLANEDRIVER",
+			Value: "/usr/local/bin/felix-plugins/felix-api-proxy",
+		}, corev1.EnvVar{
+			Name:  "FELIX_XDPENABLED",
+			Value: "false",
+		})
+		if c.cfg.Installation.KubernetesProvider == operatorv1.ProviderEKS {
+			nodeEnv = append(nodeEnv, corev1.EnvVar{
+				Name:  "FELIX_AWSSRCDSTCHECK",
```

Comment:
Hm, I was expecting us to already be setting this but it doesn't look like it. @tmjd do you recall whether we discussed this in the past?

Reply:
I don't remember talking about that one before.
I'd be concerned to default it because of this from the [reference docs](https://projectcalico.docs.tigera.io/reference/felix/configuration) 
> Check IAM role and profile configuration for setting the necessary permission for this setting to work. [Default: DoNothing]

Reply:
For this one, we've had a lot fewer issues by enabling it on EKS, and EKS nodes have the required IAM policy by default (as stated on [this page](https://projectcalico.docs.tigera.io/reference/resources/felixconfig#aws-iam-rolepolicy-for-source-destination-check-configuration)). So it would be nice to have a way to enable it. Would you prefer to expose this variable in the Installation spec and default it to `Disable` when using EKS + VPP?

Reply:
I think it's OK to leave this as it is here - if we ever do add a config option for it, we can default that option based on existing behavior.

The only consideration would be if we want to set this via `FelixConfiguration` datastore API instead of an env variable - I believe that's the path we want to be going but I can't recall if we have the right render infrastructure set up to make that easy to do.

Reply:
Sounds good to me to leave this.
As discussed in the Calico Community meeting we don't have the proper infrastructure for the FelixConfiguration to allow/understand if the users overwrote/set the field.